FutureNorthants/northants-website
# Backend support for HeroImage structure

### Setup
Checkout this branch and run a `./drush.wrapper @docker config:import` then `./drush.wrapper @docker cache-rebuild`.

### Testing
1. In the CMS - https://core-cms.northants.localhost - ensure there is now a paragraph type of Hero Image - Structure | Paragraph Types | Hero Image.
2. As I found out how to make the CMS view mode use the same CSS as the editor, I've gone back to the previous slices to apply the same to them. Edit any page to include an example of every slice available and save it. Verify that the page preview (non edit mode) now shows each of these slices in something vaguely resembling their frontend appearance, rather than just being plain text as before. Note there _isn't_ a hero image slice now.
3. Edit the page again and verify that in the editor the slices are styled similarly (there are some minor differences due to being inside the ckeditor's more comprehensive styling). Also verify it still displays OK in the frontend (should be zero differences in the graphql output!).
4. Now edit a service landing page and set up a hero image for it. Title is optional - if blank, will use the page title. Body and call to action are also optional.
5. Go to https://core-cms.northants.localhost/graphql and enter the following query, replacing the route with the route name of the service landing page you edited:

```
{
  route(path: "/health-leisure-and-parks") {
    __typename
    ... on DrupalNodeRoute {
      node {
        ... on ServiceLandingPageNode {
          __typename
          title
          heroImage {
            imageLarge
            imageSmall
            title
            body {
              value
              format
            }
            link {
              url
              title
            }
            isBoxed
            id
          }
          body {
            format
            value
          }
          hasSections {
            id
            name
          }
          id
        }
      }
    }
  }
}

```
Run the query and verify that the `heroImage` part of the result has values match those you entered in the CMS editor, or is `null` if you haven't added (or have subsequently removed) a hero image.
